contracts/Vault.vy

```diff
@@ -1100,9 +1100,9 @@ def withdraw(
             #       including the losses that were incurred above during withdrawals
             shares = self._sharesForAmount(value + totalLoss)
 
-    # NOTE: This loss protection is put in place to revert if losses from
-    #       withdrawing are more than what is considered acceptable.
-    assert totalLoss <= maxLoss * (value + totalLoss) / MAX_BPS 
+        # NOTE: This loss protection is put in place to revert if losses from
+        #       withdrawing are more than what is considered acceptable.
+        assert totalLoss * MAX_BPS <= maxLoss * (value + totalLoss)
```

Comment:
Why are these changes saving gas?

Reply:
see #353
* good /

Reply:
I don't see any benefits going from division to multiplication.

What you are pointing to in `2` is the change that moves the calculation within the if condition but I don't see a change of calculation.

On that PR only moving`totalLoss: uint256 = 0` make sense to me.

Reply:
update code   
https://github.com/yearn/yearn-vaults/blob/78c6b0d5afb4448fd40341decdd8b753a9846e90/contracts/Vault.vy#L1051
 should be below it

like #353

Reply:
I see what he's saying, but it's more clear if we keep `MAX_BPS` where it is, because it's adjusting the `maxLoss` parameter (which is really a ratio) to a value. From an auditing perspective, I prefer the clarity.

Revert this one reordering and this optimization looks good